056-zk-kraft-migration.md

```diff
@@ -0,0 +1,247 @@
+* The migration works by having ZooKeeper nodes to be replaced by new KRaft controller nodes.
+* Supporting "combined" mode is actually one of the rejected [alternatives](https://cwiki.apache.org/confluence/display/KAFKA/KIP-866+ZooKeeper+to+KRaft+Migration#KIP866ZooKeepertoKRaftMigration-CombinedModeMigrationSupport) in KIP-866.
+
+It means that the user environment should be sized to host more nodes running in parallel, because during the migration process both ZooKeeper nodes and KRaft controllers will be running.
+The end of the migration will involve shutting down the ZooKeeper nodes reducing the resources consumption.
+
+TBD: some intro about Strimzi migration process
+
+### Prerequisites
+
+An Apache Kafka cluster is running in ZooKeeper mode with `N` brokers and `M` ZooKeeper nodes, as defined by the `replicas` fields in the corresponding `Kafka` custom resource.
+The cluster has to be migrated to use `KafkaNodePool`(s) first for running the brokers (ZooKeeper nodes are never part of pools).
+Without them it’s not possible to deploy the new KRAft controllers (separated from the current Kafka brokers) as the first step of the migration.
+So the prerequisite is having the Kafka cluster still using ZooKeeper but running the Kafka brokers by using `KafkaNodePool`(s).
+The user can follow the procedure described [here](https://strimzi.io/docs/operators/latest/deploying#proc-migrating-clusters-node-pools-str) in the official documentation, in order to meet this prerequisite.
+
+Also, the minimum Kafka version should be 3.5.0 with `inter.broker.protocol.version` set to “3.5”.
+The user could need to upgrade the Kafka version on its own cluster before running a migration.
+
+### Migration
+
+The migration process is semi-automatic which means having the user driving the process by interacting with the operator (mostly via annotations and custom resources creation/deletion) which is in charge to do the rest (i.e. nodes re-configuration, rolling, ...).
+
+> NOTE: this kind of approach is already in place for the Kafka version upgrade.
+
+Assume the following prerequisites, with a ZooKeeper based Kafka cluster defined by:
+
+* A `Kafka` custom resource.
+* One or more `KafkaNodePool`(s) for running the ZooKeeper based brokers.
+
+It would assume the operator is not using KRaft by default nor has the `+UseKRaft` feature gate enabled.
+
+First step for the user would be enabling the `+UseKRaft` feature gate unless it’s already enabled by default in a new operator release (depending on which Strimzi version this proposal will land).
+So it could be needed doing an upgrade of the current operator to the new release.
+In both cases, the operator is rolled, a new one starts and it’s able to handle the creation of new Apache Kafka clusters in KRaft mode but not migrating the old ZooKeeper based ones automatically.
```

Comment:
This seems weird. You first say `UseKRaft` should be disabled. And right after that you say that it should be enabled.

Reply:
Can you point me to the right sentences you see this? Maybe I wrote bad but I meant something different :-)
The summary is that I would expect the user doesn't have KRaft already enabled so he has to enable the FG or maybe just upgrading the operator (if this migration will land in an operator version where we don't have KRaft FG anymore but it's built-in and enabled by default).

Reply:
It is right in the section I commented on. The first sentence sounds like a prerequisite saying you should have`+UseKRaft` disabled. And then right away you say to enable it.

Reply:
Ok so I think that the phrasing `It would assume the operator is not using KRaft ...` sounds like a prerequisite but it was not my intention. I will change it.